lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -404,14 +402,18 @@ defmodule Archethic.Mining.PendingTransactionValidation do
 
     {last_address, _} = DB.get_last_chain_address(genesis_address)
 
+    sorted_authorized_keys =
+      authorized_keys
+      |> Map.keys()
+      |> Enum.sort()
+
+    sorted_node_renewal_authorized_keys =
+      NodeRenewal.next_authorized_node_public_keys()
+      |> Enum.sort()
+
     with {^last_address, _} <- DB.get_last_chain_address(genesis_address, last_scheduling_date),
-         {:ok, _, _} <-
-           NodeRenewal.decode_transaction_content(content),
-         true <-
-           Enum.all?(
-             Map.keys(authorized_keys),
-             &Utils.key_in_node_list?(nodes, &1)
-           ) do
+         {:ok, _, _} <- NodeRenewal.decode_transaction_content(content),
+         true <- sorted_authorized_keys == sorted_node_renewal_authorized_keys do
```

Comment:
You may add a unit test to ensure the error when both list are different.
Otherwise it works well :+1: